### ISSUE TITLE
icon background are off at places

### ISSUE THREAD

**Issue by aasutossh** [bluz71/vim-nightfly-guicolors#35]
<img width="1440" alt="Screen Shot 2022-09-27 at 13 52 45" src="https://user-images.githubusercontent.com/18498932/192470331-d1a5a921-ad9c-47f2-9a03-91ac4569afcc.png">
for example at the top buffer tabs, the icon background are dark, but at the left file explorer icons are fine.

**Comment by bluz71** [comment]
Please provide a minimal Neovim configuration that reproduces the issue.

I suspect you are using the Bufferline plugin? I will endeavour to fix it, if possible; but to be honest that plugin has been a **major pain** to support.

**Comment by aasutossh** [comment]
But, how other themes support them well?

**Comment by bluz71** [comment]
Certain newer Neovim themes (such a Tokyonight, Catppucin and Nightfox) use very dark tab/statusline with the edit area being lighter. My themes (which are older) use a dark edit area with lighter tab/statusline. This color combination not work well with bufferline plus I find the bufferline plugin itself is overly complicated.

I wish more folks would just use [barbar.nvim](https://github.com/romgrk/barbar.nvim) instead.

Anyway, provide me a minimal setup and I will see what I can do. I suspect the issue is with NERDFonts and devicons. There probably is a solution for this particular issue.

**Comment by bluz71** [comment]
Fixed in the latest commit.

Bufferline **requires** colorscheme highlight clearing, otherwise highlight groups will be broken (and they were with my theme).

Bufferline also should be allowed to figure out the highlighting itself; it derives a many highlight groups automatically.....these should not be overriden (which I did previously, but no longer do).

Note, the bufferline is quite dark (because my theme itself is quite dark); but that's how it needs to be with this plugin.

Cheers.
